packages/components/src/modifiers/hds-clipboard.ts

```diff
@@ -125,19 +132,37 @@ export const copyToClipboard = async (text, target) => {
   return success;
 };
 
+type TextToCopy = string | number | bigint | undefined;
+type TargetToCopy = Node | undefined;
```

Comment:
I believe these abstractions make sense, as we work with them in various placed in the file.

I would try to be more specific with the types. For instance, we know that the target is always going to be an HTMLElement – this may require some adjustments across the codebase where HTMLElement should be used instead of Node (and will be highlighted in the editor), and that the text to be copied is always going to be a string (be that via `.value` or `.innerText`

```suggestion
type TextToCopy = string | undefined;
type TargetToCopy = HTMLElement | undefined;
```

Reply:
agree also with placing the types at the top of the file, so they don't get lost

Reply:
> that the text to be copied is always going to be a string

@alex-ju see here: https://github.com/hashicorp/design-system/blob/hds-3344-clipboard-ts-conversion/packages/components/src/modifiers/hds-clipboard.ts#L15-L19

Reply:
I see, so we allow numbers too – should we update [the guidance](https://helios.hashicorp.design/components/copy/button?tab=code) to reflect this? or is it more of a back up on our side?

Reply:
By the look of it, the Component API docs are not correct (outdated) too: https://helios.hashicorp.design/components/copy/button?tab=code#component-api
😢

Reply:
Opened PR to amend the documentation: https://github.com/hashicorp/design-system/pull/2121

Reply:
Thanks! Disregard my comment on `TextToCopy` then

Reply:
@alex-ju according to the getTextToCopy function, the text parameter could be a string, number, bigint, or undefined. So I undid the change to the TextToCopy types for now.

Reply:
Fixed issues with changing TargetToCopy from Node to HTMLElement type